### PR TITLE
[WIP] Refactoring

### PR DESCRIPTION
**Pull request by Schultzer**
Okay. so we need to figure out the rules we should use in our ~~`.eslintrc` and `.codeclimate.yml`~~
so we can figure out what we need to refactor and what is not necessary I need @danschultzer your 👀 on that task.

-  ~~Ruleset for `.eslintrc` and `.codeclimate.yml`~~
- [x] Removed codeclimate and `.eslintrc`
- [x] Refactoring
- [x] Indention
- [x] Single quotetation marks `'string'` for strings in our code base

### CODE REVIEW

File: cli.js [hunk 1]
```diff
@@ -49,21 +49,19 @@ function action(paths, options) {
   }
 
   if (files.length === 0) {
-    console.error('No files were found.');
-    process.exit(1);
+    throw new Error('No files were found.');
```
Comment (Schultzer) [inline cli.js]: We should throw an error instead of console.error

File: cli.js [hunk 2]
```diff
@@ -49,21 +49,19 @@ function action(paths, options) {
...
   }
 
   files = files.filter(function(element, index, array) {
     return (['pdf', 'jpeg', 'jpg', 'gif', 'png', 'bmp'].indexOf(element.split('.').pop().toLowerCase()) !== -1);
   });
 
   if (files.length === 0) {
-    console.error('No image or PDF files could be found.');
-    process.exit(1);
+    throw new Error('No image or PDF files could be found.');
```
Comment (Schultzer) [inline cli.js]: We should throw an error instead of console.error

File: cli.js [hunk 3]
```diff
@@ -199,16 +199,18 @@ function statistics(objects) {
...
-    if (!object.error) {
-      if (object.amount && object.date) {
-        ++countTotal;
-      }
-      if (object.date) {
-        ++countDate;
-      }
-      if (object.amount) {
-        ++countAmount;
+    if ({}.hasOwnProperty.call(key, objects)){
```
Comment (Schultzer) [inline cli.js]: code climate advises us that we should be careful when using objects as maps in JS http://www.2ality.com/2012/01/objects-as-maps.html @danschultzer any comment on this

File: README.md [hunk 4]
```diff
@@ -6,6 +6,7 @@ Receipt scanner extracts information from your PDF or image receipts.
 [![NPM version](https://img.shields.io/npm/v/receipt-scanner.svg)](https://www.npmjs.com/package/receipt-scanner)
```
Comment (danschultzer) [inline README.md]: Remove this, or don't add code climate badge. We should not have this many badges.

File: cli.js [hunk 5]
```diff
@@ -34,7 +34,7 @@ function action(paths, options) {
         fs.accessSync(element, fs.F_OK);
         return true;
     } catch (e) {
-      console.log("Warning: File or directory \"" + element + "\" does not exist.");
+      console.log('Warning: File or directory \'' + element + '\' does not exist.');
```
Comment (danschultzer) [inline cli.js]: Why not `console.log('Warning: File or directory "' + element + '" does not exist.');`

File: cli.js [hunk 6]
```diff
@@ -72,17 +70,21 @@ function action(paths, options) {
     stream: hideProgressBar ? function() {} : stderr
   });
 
-  var interval = setInterval(function() {
+  function updateBar() {
```
Comment (danschultzer) [inline cli.js]: Maybe rename it to `updateProgressBar`

File: cli.js (hunk 3 above)
Comment (danschultzer) [inline cli.js]: I don't see a problem when we basically use it as a hash? We don't really care if there's inherited values.

File: lib/image_processor/preprocessor/opencv-photo2scan.js [hunk 7]
```diff
@@ -39,10 +39,16 @@ function handleMatrix(im, outfile, callback, config) {
   return callback(null, outfile);
 }
 
+function drawContour(im, contours){
```
Comment (danschultzer) [inline lib/image_processor/preprocessor/opencv-photo2scan.js]: Maybe call it `verboseDrawContour`, and maybe add color option so it's more like:
```javascript
function verboseDrawContour(im, contours, color, thickness){
  color = color || [0, 255, 0]
  thickness = thickness || 5
  for(var c = 0; c < contours.size(); ++c) {
    im.drawContour(contours, c, color, thickness);
  }
}
```

File: lib/image_processor/preprocessor/sharp.js [hunk 8]
```diff
@@ -1,8 +1,8 @@
-var sh = require("sharp"),
+var sh = require('sharp'),
   fs = require('fs');
 
 function sharp(stream, outfile, cb) {
-  if (typeof stream == "string") stream = fs.createReadStream(stream);
+  if (typeof stream === 'string') stream = fs.createReadStream(stream);
```
Comment (danschultzer) [inline lib/image_processor/preprocessor/sharp.js]: Put this on a second line, or between brackets?

File: lib/pdf_processor.js [hunk 9]
```diff
@@ -24,7 +24,7 @@ PdfProcessor.prototype.process = function(file_or_stream) {
 
       // If we can't pass anything, last resort is to use tesseract
       var pages = 1;
-      if (typeof pages !== "string") pages = text.length;
+      if (typeof pages !== 'string') pages = text.length;
```
Comment (danschultzer) [inline lib/pdf_processor.js]: Same here, second line/

**Comment by danschultzer** [comment]
Thinking about this, is it actually necessary to add code climate, and not just use a linter? Code climate is a good way to identify potential issues, but we won't necessarily follow their guidelines 100%.

File: cli.js [hunk 10]
```diff
@@ -72,17 +70,21 @@ function action(paths, options) {
...
+  var interval = setInterval(function() {
+    updateProgressBar();
 
     if (bar.complete) clearInterval(interval);
   }, 300);
 
   var filesDone = [];
 
-  for (i = 0; i < files.length; i++) {
+  for (let i = 0; i < files.length; i++) {
```
Comment (danschultzer) [inline cli.js]: Use `var` and `j` instead of `i` here. The rest of this code base is not ECMA6.

File: lib/text_parser/amount_parser.js [hunk 11]
```diff
@@ -7,10 +7,14 @@ function amountParser(text, config) {
...
   for(var i = 0; i < parsers.length; i++) {
     parser = parsers[i];
-    if (typeof parser === "string") parser = require('./amount/' + parser);
+    if (typeof parser === "string") {
+      parser = require('./amount/' + parser);
+    }
     result.matches = allAmounts(text, parser.getOpts ? parser.getOpts() : {});
     match = parser.extract(result.matches);
-    if (match) break;
+    if (match) {
```
Comment (danschultzer) [inline lib/text_parser/amount_parser.js]: This is probably better as:
```javascript
if (match)
  break;
```

**Comment by danschultzer** [comment]
Overall it looks fine, but some of the if statements should just be without curly brackets.

File: benchmark/image_preprocessors.js [hunk 12]
```diff
@@ -42,29 +42,28 @@ function benchmark(preprocessors, compareFiles) {
...
+    log('Working in ' + chalk.underline(benchmarkDir));
     fs.access(saveAs, fs.F_OK, function(error) {
-      if (!error) {
+      if (!error)
         return resolve(saveAs);
-      }
-      log("Downloading " + chalk.underline("receipt-scanner-testdata-master.zip"), " ...");
-      var command = "curl -o " + saveAs + " -z " + saveAs + " -L https://github.com/danschultzer/receipt-scanner-testdata/archive/master.zip";
-      log("Running " + chalk.bold(command));
+      log('Downloading ' + chalk.underline('receipt-scanner-testdata-master.zip'), ' ...');
```
Comment (danschultzer) [inline benchmark/image_preprocessors.js]: Should have newline here

**Pull request merged**